crates/gosub_net/src/http/headers.rs

```diff
@@ -12,10 +12,20 @@ impl Headers {
         }
     }
 
-    pub fn set(&mut self, key: &str, value: &str) {
+    pub fn with_capacity(capacity: usize) -> Headers {
+        Headers {
+            headers: HashMap::with_capacity(capacity),
+        }
+    }
+
+    pub fn set_str(&mut self, key: &str, value: &str) {
```

Comment:
Hello hello, once again a random comment after long silence 😉 I am actually very actively following the activity here on GitHub, but don't have a lot of time for communication. This has already been merged, but a little note for the future: Rust allows you to use `impl Trait` types as parameters, so you don't need separate methods `set` and `set_str`. That is widely used in the standard library too. The method would look like this:
```rs
pub fn set(&mut self, key: impl AsRef<str>, value: impl AsRef<str>) {
    self.headers.insert(key.as_ref().to_string(), value.as_ref().to_string());
}
```
When calling it with owned strings, the compiler will completely optimize this away into just `self.headers.insert(key, value);`. Because `impl` is just syntax sugar for bounded generics, the method actually gets monomorphized, resulting in the same 2 methods, but you are able to "generate" more by calling it with `&str, String` for example.
```rs
req.header.set("Content-Type", format!("text/{extension}")); // bad, don't do this, just an example
```

Reply:
Ah, yeah. That looks interesting. I guess we could also just directly use `impl Into<String>`. I might change that when I'm adding more functionality to the fetcher. But thanks for the suggestion!